AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt

```diff
@@ -0,0 +1,702 @@
+            state = lazyListState
+        ) {
+            // Weather
+            item { Header(title = stringResource(R.string.weather)) }
+            item { Spacer(modifier = Modifier.height(16.dp)) }
+            item {
+                Surface(
+                    modifier = Modifier.fillMaxWidth(),
+                    elevation = 2.dp
+                ) {
+                    if (weatherLoading) {
+                        LoadingRow()
+                    } else {
+                        WeatherRow(onRefresh = {
+                            scope.launch {
+                                loadWeather()
+                            }
+                        })
+                    }
+                }
+            }
+            item { Spacer(modifier = Modifier.height(32.dp)) }
+
+            // Topics
+            item { Header(title = stringResource(R.string.topics)) }
+            item { Spacer(modifier = Modifier.height(16.dp)) }
+            items(count = allTopics.size) { i ->
+                val topic = allTopics[i]
+                TopicRow(
+                    topic = topic,
+                    expanded = expandedTopic == topic,
+                    onClick = {
+                        expandedTopic = if (expandedTopic == topic) null else topic
+                    }
+                )
+            }
+            item { Spacer(modifier = Modifier.height(32.dp)) }
+
+            // Tasks
+            item { Header(title = stringResource(R.string.tasks)) }
+            item { Spacer(modifier = Modifier.height(16.dp)) }
+            if (tasks.isEmpty()) {
+                item {
+                    TextButton(onClick = { tasks.clear(); tasks.addAll(allTasks) }) {
+                        Text(stringResource(R.string.add_tasks))
+                    }
+                }
+            }
+            items(count = tasks.size) { i ->
+                val task = tasks.getOrNull(i)
+                if (task != null) {
+                    key(task) {
+                        TaskRow(
+                            task = task,
+                            onRemove = { tasks.remove(task) }
+                        )
+                    }
+                }
+            }
+        }
+        EditMessage(editMessageShown)
+    }
+}
+
+/**
+ * Shows the floating action button.
+ *
+ * @param extended Whether the tab should be shown in its expanded state.
+ */
+// AnimatedVisibility is currently an experimental API in Compose Animation.
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun HomeFloatingActionButton(
+    extended: Boolean,
+    onClick: () -> Unit
+) {
+    FloatingActionButton(onClick = onClick) {
+        Row(
+            modifier = Modifier.padding(horizontal = 16.dp)
+        ) {
+            Icon(
+                imageVector = Icons.Default.Edit,
+                contentDescription = stringResource(R.string.edit)
```

Comment:
Shouldn't CD be null as this will be merged with the Text element as it's a single clickable?

Reply:
Yes, this should be `null`. The FloatingActionButton is clickable, which means it merges the content descriptions of the descendants. The icon is decorative in this sense.

Reply:
Done